src/extensions/agent_modules/cloud_communication/eg_mqtt_broker_client/src/upd_request_operation.c

```diff
@@ -481,13 +484,14 @@ static void HandleReportResults(ADUC_Retriable_Operation_Context* context, ADUC_
                 free(updData->reportingJson);
                 updData->reportingJson = NULL;
                 ADUC_StateStore_SetReportResultsAck(false);
-                AduUpdUtils_TransitionState(ADU_UPD_STATE_REPORT_RESULTS_ACK, updData);
+                AduUpdUtils_TransitionState(ADU_UPD_STATE_REPORT_RESULTS_ACK, updData, context);
```

Comment:
in line freeing of updData reportingJson is being freed inline here. Do we need to free it if the SendUpdateResultsRequest fails? 

Also using context->retryParams here for the maxDelaySecs but max_retries doesn't use retry params. See line 459: 

> const unsigned int max_retries = context->retryParams[ADUC_RETRY_PARAMS_INDEX_DEFAULT].maxRetries